sui_core/src/authority/authority_store.rs

```diff
@@ -990,7 +1004,10 @@ impl<const ALL_OBJ_VER: bool> BackingPackageStore for SuiDataStore<ALL_OBJ_VER>
     }
 }
 
-impl<const ALL_OBJ_VER: bool> ModuleResolver for SuiDataStore<ALL_OBJ_VER> {
+impl<const A: bool, S: AuthoritySignInfoTrait> ModuleResolver for SuiDataStore<A, S>
```

Comment:
How about replacing `AuthoritySignInfoTrait` with its traits (which I suspect may be a bit too strong: my hunch is `Eq + Serialize + for<'de> Deserialize<'de>` would work)?